client/my-sites/upgrades/domain-search/domain-search.jsx

```diff
@@ -41,10 +41,14 @@ var DomainSearch = React.createClass( {
 	componentDidMount: function() {
 		this.props.sites.on( 'change', this.checkSiteIsUpgradeable );
 		this.props.fetchSitePlans( this.props.sitePlans, this.props.sites.getSelectedSite() );
+
+		this.previousSelectedSite = this.props.sites.getSelectedSite();
 	},
 
 	componentWillReceiveProps: function() {
-		this.props.fetchSitePlans( this.props.sitePlans, this.props.sites.getSelectedSite() );
+		if ( this.previousSelectedSite !== this.props.sites.getSelectedSite() ) {
+			this.props.fetchSitePlans( this.props.sitePlans, this.props.sites.getSelectedSite() );
+		}
```

Comment:
I think `this. previousSelectedSite` should be updated with the new selectedSite here